custom/mgh_epic/sync_epic_appointments.py

```diff
@@ -0,0 +1,321 @@
+
+        # Update existing appointments in commcare if properties have changed in epic
+        for appointment in epic_appointments_to_update:
+            epic_properties_map = {}
+            appointment_resource = appointment.get('resource')
+            if appointment_resource is not None:
+                appointment_description = appointment_resource.get('description') or 'NO DESCRIPTION LISTED'
+                appointment_fhir_timestamp = appointment_resource.get('start')
+                appointment_date, appointment_time = convert_utc_timestamp_to_date_and_time(
+                    appointment_fhir_timestamp)
+                appointment_fhir_id = appointment_resource.get('id')
+                reason = ''
+                practitioner = ''
+                for p in appointment_resource.get('participant'):
+                    actor = p.get('actor')
+                    if actor and actor.get('reference') is not None and 'Practitioner' in actor.get('reference'):
+                        practitioner = actor.get('display')
+                        break
+                reason_code = appointment_resource.get('reasonCode')
+                if reason_code and reason_code[0] is not None:
+                    reason = reason_code[0].get('text')
+                epic_properties_map.update({
+                    'appointment_description': appointment_description,
+                    'appointment_fhir_timestamp': appointment_fhir_timestamp,
+                    'practitioner': practitioner,
+                    'reason': reason
+                })
+            appointment_case = appointment_map.get(appointment_fhir_id)
+            appointment_update_helper = CaseHelper(case=appointment_case, domain=domain)
+            case_properties_to_update = {}
+            changes = False
+            # check for changes and add to case_properties_to_update
+            for k, v in epic_properties_map.items():
+                current_value = appointment_case.get_case_property(k)
+                if current_value != v:
+                    if k == 'appointment_fhir_timestamp':
+                        # these are quivalent to the minute
+                        if v[0:16] == current_value[0:16]:
+                            continue
+                        else:
+                            appointment_date, appointment_time = convert_utc_timestamp_to_date_and_time(v)
+                            case_properties_to_update.update({
+                                'appointment_date': appointment_date,
+                                'appointment_time': appointment_time,
+                            })
+                    changes = True
+                    case_properties_to_update.update({k: v})
+
+            if changes:
+                appointment_update_helper.update({'properties': case_properties_to_update})
+
+
+MGH_EPIC_DOMAINS = settings.CUSTOM_DOMAINS_BY_MODULE['custom.mgh_epic']
```

Comment:
would a feature flag not be better for this?

Reply:
It was originally under a FF, but I moved it on Norman's suggestion. 
"...if this code is specific to only one project, it should go in custom.epic?"